masonry/src/contexts.rs

```diff
@@ -361,86 +361,36 @@ impl<'a> WidgetCtx<'a> {
             widget_children: child_mut.children,
         };
         WidgetMut {
             ctx: child_ctx,
             widget: child_mut.item.as_mut_dyn_any().downcast_mut().unwrap(),
-            is_reborrow: false,
         }
     }
-}
 
-// TODO - It's not clear whether EventCtx should be able to create a WidgetMut.
-// One of the examples currently uses that feature to change a child widget's color
-// in reaction to mouse events, but we might want to address that use-case differently.
-impl<'a> EventCtx<'a> {
-    /// Return a [`WidgetMut`] to a child widget.
-    pub fn get_mut<'c, Child: Widget>(
-        &'c mut self,
-        child: &'c mut WidgetPod<Child>,
-    ) -> WidgetMut<'c, Child> {
-        let child_state_mut = self
-            .widget_state_children
-            .get_child_mut(child.id().to_raw())
-            .expect("get_mut: child not found");
-        let child_mut = self
-            .widget_children
-            .get_child_mut(child.id().to_raw())
-            .expect("get_mut: child not found");
-        let child_ctx = WidgetCtx {
-            global_state: self.global_state,
-            parent_widget_state: self.widget_state,
-            widget_state: child_state_mut.item,
-            widget_state_children: child_state_mut.children,
-            widget_children: child_mut.children,
-        };
-        WidgetMut {
-            ctx: child_ctx,
-            widget: child_mut.item.as_mut_dyn_any().downcast_mut().unwrap(),
-            is_reborrow: false,
-        }
-    }
-}
-
-// TODO - It's not clear whether LifeCycleCtx should be able to create a WidgetMut.
-impl<'a> LifeCycleCtx<'a> {
-    /// Return a [`WidgetMut`] to a child widget.
-    pub fn get_mut<'c, Child: Widget>(
-        &'c mut self,
-        child: &'c mut WidgetPod<Child>,
-    ) -> WidgetMut<'c, Child> {
-        let child_state_mut = self
-            .widget_state_children
-            .get_child_mut(child.id().to_raw())
-            .expect("get_mut: child not found");
-        let child_mut = self
-            .widget_children
-            .get_child_mut(child.id().to_raw())
-            .expect("get_mut: child not found");
-        let child_ctx = WidgetCtx {
+    pub(crate) fn reborrow_mut(&mut self) -> MutateCtx<'_> {
+        MutateCtx {
             global_state: self.global_state,
-            parent_widget_state: self.widget_state,
-            widget_state: child_state_mut.item,
-            widget_state_children: child_state_mut.children,
-            widget_children: child_mut.children,
-        };
-        WidgetMut {
-            ctx: child_ctx,
-            widget: child_mut.item.as_mut_dyn_any().downcast_mut().unwrap(),
-            is_reborrow: false,
+            // We don't don't reborrow parent_widget_state. This means a WidgetMut
+            // made from this won't merge state on Drop, which is usually what you want
+            // from a reborrowed WidgetMut.
```

Comment:
Maybe something like this would be clearer.
```suggestion
            // We don't don't reborrow `parent_widget_state`. This avoids running
            // `merge_up` in `Drop` multiple times for this widget (it will still be called when
            // `self` is dropped)
```